src/PrettierEditProvider.ts

```diff
@@ -110,23 +113,41 @@ function format(
     );
 
     if (config.eslintIntegration && !isNonJsParser) {
-        const prettierEslint = require('prettier-eslint') as PrettierEslintFormat;
-        return prettierEslint({
+        return safeExecution(
+            () => {
+                const prettierEslint = require('prettier-eslint') as PrettierEslintFormat;
+                return prettierEslint({
+                    text,
+                    filePath: fileName,
+                    fallbackPrettierOptions: prettierOptions,
+                });
+            },
             text,
-            filePath: fileName,
-            fallbackPrettierOptions: prettierOptions,
-        });
+            fileName
+        );
     }
     const prettier = requireLocalPkg(fileName, 'prettier') as Prettier;
     if (isNonJsParser && !parserExists(parser, prettier)) {
-        const bundledPrettier = require('prettier') as Prettier;
-        window.showWarningMessage(
-            `prettier@${prettier.version} doesn't support ${languageId}. ` +
-                `Falling back to bundled prettier@${bundledPrettier.version}.`
+        return safeExecution(
+            () => {
+                const bundledPrettier = require('prettier') as Prettier;
+                const warningMessage =
+                    `prettier@${prettier.version} doesn't support ${languageId}. ` +
+                    `Falling back to bundled prettier@${bundledPrettier.version}.`;
+                window.showWarningMessage(warningMessage);
```

Comment:
You are right.
Dummy implementation:
let a boolean 'errorShown' = false
don't show that error if errroShown
set it to true once shown.
set it to false on workspace change event. (I pretty sure that one exists)